ubclaunchpad/ubc-degree-navigator
# Added patch completedCourses

## Describe your changes
> Add a short description of the changes. These will be the things to check for when the reviewers review your PR.
Added patch request for completedcourses. I made the assumption that we should look for completedcourses based on their userID and courseID (ie they are the primary key), so that's the criteria for which a course is updated on. 200 status code when successfully updated, 404 when entry doesn't exist, 500 when failed to update.

I also slightly modified the code for post completedcourses; namely, I removed a duplicate type struct for completedcourses, changed the parameter list of LoadCompletedCourseToDB to just a completedcourse pointer rather than all the fields being passed in individually, and also made it so that post completedcourse returns the completedcourse that was created. Let me know if any of this should be changed.

#### Screenshot (if applicable)
> This can be FE screenshot, DB screenshot, API call from network tab, etc.
![Screenshot 2023-12-07 205544](https://github.com/ubclaunchpad/ubc-degree-navigator/assets/111270733/2a06a5d5-d001-47ef-8080-7f9cebf5664d)
![Screenshot 2023-12-07 205611](https://github.com/ubclaunchpad/ubc-degree-navigator/assets/111270733/ae13d14e-620f-4475-8525-bdd5d1c0d9f8)
You can see that in the second screenshot, the credit_counted value has changed to 6 from 3.

## How do I test
> Provide brief steps on how to test these changes. 
1. Create a completedcourse row in the DB (I did this via the post request).
2. Make a patch request, passing in the courseid and userid of the completedcourse you want to update, and pass in the field-value pairs you want to have updated in the request body. You only need to pass in the pairs that you want to have changed (ie if you only want to change sessioncompleted, then you pass in courseid, userid, and sessioncompleted).
3. If you use DB Browser for SQLite and load the gorm.db file onto it, you'll see that in the completed courses table, the row has been updated.
4. Also make sure that the appropriate data and status codes are returned after a request

## Checklist before requesting a review
